x-pack/plugins/osquery/server/lib/telemetry/helpers.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { SavedObjectsFindResponse } from 'kibana/server';
+import { PackagePolicy } from '../../../../fleet/common/types/models/package_policy';
+import { copyAllowlistedFields, packEventFields, savedQueryEventFields } from './filters';
+import type { ESClusterInfo, ESLicense, ListTemplate, TelemetryEvent } from './types';
+
+/**
+ * Determines the when the last run was in order to execute to.
+ *
+ * @param executeTo
+ * @param lastExecutionTimestamp
+ * @returns the timestamp to search from
+ */
+export const getPreviousDiagTaskTimestamp = (
```

Comment:
This is alert diagnostic telemetry-specific. Also, warning - this code needs a lot of work so please proceed with caution.